crates/core/src/kakarot/mod.rs

```diff
@@ -0,0 +1,123 @@
+use std::marker::PhantomData;
+
+use starknet::core::types::{BlockId, FunctionCall};
+use starknet::providers::Provider;
+use starknet_crypto::FieldElement;
+
+use crate::client::constants::selectors::{BALANCE_OF, BYTECODE, COMPUTE_STARKNET_ADDRESS, ETH_CALL, GET_EVM_ADDRESS};
+use crate::client::constants::STARKNET_NATIVE_TOKEN;
+use crate::client::errors::EthApiError;
+use crate::client::helpers::DataDecodingError;
+
+pub struct KakarotContract<P> {
+    pub address: FieldElement,
+    pub proxy_account_class_hash: FieldElement,
+    _phantom: PhantomData<P>,
+}
+
+impl<P: Provider + Send + Sync> KakarotContract<P> {
```

Comment:
let's make this `impl` returned parsed results so we don't do conversation outside of the contract.